docs: add instructions for connecting edge host to wifi (#2092)
spectrocloud/librarium: docs/docs-content/clusters/edge/networking/connect-wifi.md

### Prerequisites

- A physical or virtual Linux machine with AMD64 (also known as x86_64) processor architecture to build the Edge
  artifacts. You can issue the following command in the terminal to check your processor architecture.
- Minimum hardware configuration of the Linux machine:
  - 4 CPU
  - 8 GB memory
  - 50 GB storage
- A bootable device such as a USB drive, or a Preboot Execution Environment (PXE) server.
- A [Spectro Cloud](https://console.spectrocloud.com/) account. If you have not signed up, you can sign up for a
  [free trial](https://www.spectrocloud.com/free-tier/).
- Palette registration token for pairing Edge hosts with Palette. You will need tenant admin access to Palette to
  generate a new registration token. For detailed instructions, refer to the Create Registration Token guide.

### Procedure

1. Clone the CanvOS repository.

   ```shell
   git clone https://github.com/spectrocloud/CanvOS.git
   ```

2. Edit the file **Dockerfile** in the root directory of the repository. At the bottom of the file, add the following
   lines.

   ```
   RUN apt-get update && apt-get install wpasupplicant && \
       mkdir /var/lib/wpa
   ```

   :::tip

>    Consider including some other lightweight network utilities in the OS image as well that can help you troubleshoot or

Reviewer: ```suggestion
   Consider including some other lightweight network utilities in the OS image that can help you troubleshoot or
```

>    set up network configurations alternatively if needed. For example, replace the RUN statement with the following to

Reviewer: ```suggestion
   set up network configurations alternatively if needed. For example, replace the RUN statement with the following commands to
```

   also include network manager and ping.

   ```
   RUN apt-get update && apt-get install wpasupplicant -y && \
    apt-get update && apt-get install network-manager -y && \
    apt-get install iputils-ping -y && \
    mkdir /var/lib/wpa
   ```

   :::

> 3. Follow the rest of the steps in the guide Build Edge Artifact. When finished, identify the

Reviewer: Missing link for Build Edge Artifact guide?

   **palette-edge-installer.iso** file in the **build** folder.
4. Plug in your bootable device and flash it with the ISO image.

> 5. Plug the bootable device into the Edge host and power on the Edge host. When the Edge host powers on, open the boot

Reviewer 2: Should menu be Menu? I'll defer to you Lenny.

   menu by pressing F10, and select the bootable device as your boot volume.

> 6. Wait for installation to finish. The Edge host will reboot by default upon completion unless you specify a different

Reviewer: ```suggestion
6. Wait for the installation to finish. The Edge host will reboot by default upon completion unless you specify a different
```

   option in the Edge Installer configuration user data.

### Validate

You can validate that the Edge host is ready for the site installation by powering on the Edge host and simulating a
site deployment. The simulation process will require you to complete the installation process and reset the device after
the validation.

## Supply Site-Specific User Data

> It's likely that your Wi-Fi network name and passwords are site-specific, which means you won't be able to include the

Reviewer: ```suggestion
It is likely that your Wi-Fi network name and passwords are site-specific, which means you won't be able to include the
```

network credentials in the installer. You can use site-specific user data to provide this information to you Edge hosts
instead.

### Prerequisites

- A bootable device such as a USB drive, or a Preboot Execution Environment (PXE) server.
- `mkisofs`, or `genisoimage`, or similar ISO management software.
  - `cdrtools` or `wodim` for Windows.
- An Edge host that has already been prepared for installation.

### Procedure

1. Create a file named `user-data`, and include the following content. Replace `network-name` and `network-password`

>    with the name and password of your Wifi network.

Reviewer: ```suggestion
   with the name and password of your Wi-Fi network.
```

   ```yaml
    #cloud-config
    install:
        bind_mounts:
            - /var/lib/wpa

    stages:
        network.before:
        - name: "Connect to Wi-Fi"
        commands:
            - |
            # Find the first wireless network interface
            wireless_interface=""
            for interface in $(ip link | grep -oP '^\d+: \K[^:]+(?=:)')
            do
                if [ -d "/sys/class/net/$interface/wireless" ]; then
                    wireless_interface=$interface
                    break
                fi
            done

            # Check if a wireless interface was found and connect it to WiFi
            if [ -n "$wireless_interface" ]; then
                wpa_passphrase <network-name> <network-password> | tee /var/lib/wpa/wpa_supplicant.conf
                wpa_supplicant -B -c /var/lib/wpa/wpa_supplicant.conf -i $wireless_interface
                dhclient $wireless_interface
            else
                echo "No wireless network interface found."
            fi
   ```

> 2. Follow the rest of the steps in [Apply Site User Data](../site-deployment/site-installation/site-user-data.md).

Reviewer: ```suggestion
2. Follow the rest of the steps in the [Apply Site User Data](../site-deployment/site-installation/site-user-data.md) section.
```

### Validate

> 1. Power up the Edge host and wait for it to boot to the registration screen. Do not connect the Edge host with a wired

Reviewer: Maybe a call out box for **Do not connect the Edge host with a wired**?
